packages/vscode-extension/src/devices/preview.ts

```diff
@@ -38,27 +44,95 @@ export class Preview implements Disposable {
         reject(new Error("Preview server exited without URL"));
       });
 
-      const streamURLRegex = /(http:\/\/[^ ]*stream\.mjpeg)/;
-
       lineReader(subprocess).onLineRead((line, stderr) => {
         if (stderr) {
           Logger.info("sim-server:", line);
           return;
         }
 
-        const match = line.match(streamURLRegex);
+        if (line.includes("stream_ready")) {
+          const streamURLRegex = /(http:\/\/[^ ]*stream\.mjpeg)/;
+          const match = line.match(streamURLRegex);
+
+          if (match) {
+            Logger.info(`Stream ready ${match[1]}`);
 
-        if (match) {
-          Logger.info(`Stream ready ${match[1]}`);
+            this.streamURL = match[1];
+            resolve(this.streamURL);
+          }
+        } else if (line.includes("video_ready replay") || line.includes("video_error replay")) {
+          // video response format for replays looks as follows:
+          // video_ready replay <HTTP_URL> <FILE_URL>
+          // video_error replay <Error message>
+          const videoReadyMatch = line.match(/video_ready replay\S+ (\S+) (\S+)/);
+          const videoErrorMatch = line.match(/video_error replay\S+ (.*)/);
 
-          this.streamURL = match[1];
-          resolve(this.streamURL);
+          const handlers = this.lastReplayPromise;
+          this.lastReplayPromise = undefined;
+
+          if (handlers && videoReadyMatch) {
+            // match array looks as follows:
+            // [0] - full match
+            // [1] - URL or error message
+            // [2] - File URL
+            const tempFileLocation = videoReadyMatch[2];
+            const ext = path.extname(tempFileLocation);
+            const fileName = workspace.name
+              ? `${workspace.name} RadonIDE replay${ext}`
+              : `RadonIDE-replay${ext}`;
```

Comment:
We should separate words with dashes in filename also when condition is true
```suggestion
              ? `${workspace.name}-RadonIDE-replay${ext}`
              : `RadonIDE-replay${ext}`;
```